pyproject.toml

```diff
@@ -1,6 +1,187 @@
+    "ipywidgets >= 7.5",
+    "xgboost >= 1.5.1",
+    "catboost >= 0.20",
+    "lightgbm >= 2.3.1",
+    "matplotlib >= 3.3.3",
+    "graphviz >= 0.13; platform_system!='Windows'",
+    "seaborn >= 0.11.1",
+    "category_encoders >= 2.2.2",
+    "imbalanced-learn >= 0.9.1",
+    "pmdarima >= 1.8.1, < 2.0.0",
+    "sktime >= 0.13.3",
+    "lime >= 0.2.0.1",
+    "vowpalwabbit >= 8.11.0",
+]
+
+[project.urls]
+"Documentation" = "https://evalml.alteryx.com"
+"Source Code"= "https://github.com/alteryx/evalml/"
+"Changes" = "https://evalml.alteryx.com/en/latest/release_notes.html"
+"Issue Tracker" = "https://github.com/alteryx/evalml/issues"
+"Twitter" = "https://twitter.com/alteryxoss"
+"Chat" = "https://join.slack.com/t/alteryx-oss/shared_invite/zt-182tyvuxv-NzIn6eiCEf8TBziuKp0bNA"
+
+[project.optional-dependencies]
+test = [
+    "pytest == 7.1.2",
+    "pytest-xdist == 2.1.0",
+    "pytest-timeout == 1.4.2",
+    "pytest-cov == 2.10.1",
+    "nbval == 0.9.3",
+    "IPython >= 7.16.3",
+    "codecov == 2.1.11",
+    "PyYAML == 5.4",
+    "coverage[toml] >= 6.4",
+    "tomli >= 2.0.1",
+]
+dev = [
+    "ruff >= 0.0.219",
+    "black[jupyter] >= 22.3.0",
+    "darglint == 1.8.0",
+    "pre-commit >= 2.20.0",
+    "evalml[docs, test]",
+]
+updater = [
+    "alteryx-open-src-update-checker >= 2.1.0"
+]
+docs = [
+    "docutils >=0.15.2, <0.17",
+    "pydata-sphinx-theme >= 0.3.1",
+    "astroid <= 2.6.6",
+    "Sphinx >= 4.5.0",
+    "nbconvert >= 6.4.5",
+    "nbsphinx >= 0.8.5",
+    "sphinx-autoapi",
+    "sphinx-inline-tabs == 2022.1.2b11",
+    "sphinx-copybutton == 0.4.0",
+    "myst-parser == 0.16.1",
+]
+prophet = [
+    "prophet-prebuilt == 1.0.2",
+]
+complete = [
+    "evalml[prophet, update_checker]",
+]
+
+[tool.setuptools]
+include-package-data = true
+license-files = ["LICENSE"]
+
+[tool.setuptools.packages.find]
+namespaces = true
+
+[tool.setuptools.package-data]
+"*" = [
+    "*.txt",
+    "README.md",
+]
+"evalml.tests.data" = [
+    "evalml/tests/data/churn.csv",
+    "evalml/tests/data/daily-min-temperatures.csv",
+    "evalml/tests/data/fraud_transactions.csv.gz",
+    "evalml/tests/data/tips.csv",
+    "evalml/tests/data/titanic.csv",
+]
+
+[tool.setuptools.exclude-package-data]
+"*" = [
+    "* __pycache__",
+    "*.py[co]",
+    "docs/*"
+]
+
+[tool.setuptools.dynamic]
+version = {attr = "evalml.__version__"}
+
+[tool.ruff]
+ignore = ["E501"]
+select = [
+    # Pyflakes
+    "F",
+    # Pycodestyle
+    "E",
+    "W",
+    # isort
+    "I001"
+]
+src = ["evalml"]
+
+[tool.ruff.per-file-ignores]
+"__init__.py" = ["E402", "F401", "I001", "E501"]
+
+[tool.ruff.isort]
+known-first-party = ["evalml"]
+
+[tool.coverage.run]
+source = ["evalml"]
+
+[tool.coverage.report]
+exclude_lines =[
+    "pragma: no cover",
+    "def __repr__",
+    "raise AssertionError",
+    "raise NotImplementedError",
+    "if __name__ == .__main__.:",
+    "if self._verbose:",
+    "if verbose:",
+    "if profile:",
+    "pytest.skip"
```

Comment:
Can we remove the unnecessary ones here? We do really want to be making sure we're testing things like verbosity and errors.

Reply:
Done:
https://github.com/alteryx/evalml/pull/3928/commits/98aff8f20d4050b9f809f713bad61434e1ffd755#diff-50c86b7ed8ac2cf95bd48334961bf0530cdc77b5a56f852c5c61b89d735fd711L172